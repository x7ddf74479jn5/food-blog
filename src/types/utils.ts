export type TCategoryOrTagColor = "green" | "blue" | "indigo" | "purple" | "red" | "yellow" | "pink" | "gray";

export type TImage = {
  url: string;
  width: number;
  height: number;
  blurDataURL?: string;
};

export type TImageOption = {
  fontColor: string;
};

export type ValueOf<T> = T[keyof T];

export type MappedConst<T extends string> = {
  [K in T]: K;
};
