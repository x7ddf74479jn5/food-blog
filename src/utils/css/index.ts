export { classNames } from "./classNames";
export { getBGColor, getBorderColor } from "./getColor";
