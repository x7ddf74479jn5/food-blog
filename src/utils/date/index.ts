export { formatJpYYYYMD } from "./format";
export { getNewDate } from "./getNewDate";
export { getSafeDate } from "./getSafeDate";
export { isValidISODate } from "./is-valid";
