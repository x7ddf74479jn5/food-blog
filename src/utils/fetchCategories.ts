import { client } from "src/lib/client";
import type { TCategoryListResponse } from "src/types";

type Args = {
  slug?: string;
};

export const fetchCategories = async (props?: Args) => {
  const { slug } = props || {};

  const data = await client.get<TCategoryListResponse>({ endpoint: "categories" });

  if (slug) {
    return data.contents.find((item) => item.slug === slug);
  }
  return data.contents;
};

export default fetchCategories;
