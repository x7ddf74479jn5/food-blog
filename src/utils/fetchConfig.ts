import { client } from "lib/client";
import type { TConfig } from "src/types";

export const fetchConfig = async () => {
  const data = await client
    .get<TConfig>({ endpoint: "configs" })
    .then((res) => res)
    .catch(() => null);

  return data;
};

export default fetchConfig;
