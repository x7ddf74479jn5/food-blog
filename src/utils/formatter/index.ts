export { formatPageTitle, formatPageUrl } from "./formatPage";
export { getBGColor } from "./getColor";
export { getExcerpt } from "./getExcerpt";
