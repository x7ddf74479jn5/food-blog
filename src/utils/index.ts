export const toIdleTask = (callback: () => void) => {
  if ("requestIdleCallback" in window) {
    requestIdleCallback(callback);
  } else {
    setTimeout(callback, 0);
  }
};
