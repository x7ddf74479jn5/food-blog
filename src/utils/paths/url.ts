export const apiRoute = {
  apiPreview: "/api/preview",
  apiArticles: "/api/articles",
} as const;

export const urlTable = {
  home: "/",
  root: "/",
  articles: "/articles",
  categories: "/articles/categories",
  tags: "/articles/tags",
  preview: "/preview",
  search: "/search",
} as const;

const labelTable = {
  [urlTable.home]: "レシピ一覧へ",
  [urlTable.categories]: "カテゴリー別一覧へ",
  [urlTable.tags]: "タグ別一覧へ",
} as const;

type LabelTableKey = keyof typeof labelTable;

export const getBackLinks = (paths: LabelTableKey[]): { href: string; label: string }[] => {
  return paths.map((path) => {
    return { href: path, label: labelTable[path] };
  });
};
