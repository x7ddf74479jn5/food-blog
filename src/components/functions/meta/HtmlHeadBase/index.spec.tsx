import { render } from "jest/test-utils";
import { mockArticles, mockConfig } from "mocks/data";

import { formatPageTitle, formatPageUrl } from "@/utils/formatter";
import { urlTable } from "@/utils/paths/url";

import { HtmlHeadBase } from ".";

jest.mock("next/head", () => {
  return {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    __esModule: true,
    default: ({ children }: { children: Array<React.ReactElement> }) => {
      return <>{children}</>;
    },
  };
});

describe("components/atoms/meta/HtmlHeadBase", () => {
  const { description, id, image, title } = mockArticles.stock;
  const imageUrl = image.url;
  const { host: indexUrl, siteTitle } = mockConfig;
  const pageTitle = formatPageTitle(title, siteTitle);
  const url = formatPageUrl(`${urlTable.articles}/${id}`, indexUrl);

  it("OK: 出力結果が正しい", () => {
    const { container } = render(
      <HtmlHeadBase indexUrl={indexUrl} pageTitle={pageTitle} url={url} description={description} image={imageUrl} />
    );
    expect(container.querySelector("title")).toHaveTextContent(pageTitle);
    expect(container.querySelector('meta[name="robots"]')?.attributes.getNamedItem("content")?.value).toBe(
      "index,follow"
    );
    // expect(container.querySelector('meta[name="googlebot"]')?.attributes.getNamedItem("content")?.value).toBe(
    //   "index,follow"
    // );
    expect(container.querySelector('meta[name="description"]')?.attributes.getNamedItem("content")?.value).toBe(
      description
    );
    expect(container.querySelector('meta[property="og:url"]')?.attributes.getNamedItem("content")?.value).toBe(url);
    expect(container.querySelector('meta[property="og:title"]')?.attributes.getNamedItem("content")?.value).toBe(
      pageTitle
    );
    expect(container.querySelector('meta[property="og:image"]')?.attributes.getNamedItem("content")?.value).toBe(
      imageUrl
    );
    expect(container.querySelector('meta[property="og:description"]')?.attributes.getNamedItem("content")?.value).toBe(
      description
    );
    expect(container.querySelector('link[rel="canonical"]')?.attributes.getNamedItem("href")?.value).toBe(url);
    expect(container.querySelector('link[rel="alternate"]')?.attributes.getNamedItem("href")?.value).toBe("/feed.xml");
    expect(container.querySelector('link[rel="manifest"]')?.attributes.getNamedItem("href")?.value).toBe(
      "/favicon/site.webmanifest"
    );
  });
});
