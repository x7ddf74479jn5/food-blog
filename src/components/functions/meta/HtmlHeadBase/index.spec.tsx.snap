// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/meta/HtmlHeadBase snapshot 1`] = `
Array [
  <title>
    基本の一番だしの作り方 | Food Blog
  </title>,
  <meta
    content="index,follow"
    name="robots"
  />,
  <meta
    content="だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。"
    name="description"
  />,
  <meta
    content="基本の一番だしの作り方 | Food Blog"
    property="og:title"
  />,
  <meta
    content="だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。"
    property="og:description"
  />,
  <meta
    content="https://food-blog-chi.vercel.app/articles/p97vmuno3jdn"
    property="og:url"
  />,
  <meta
    content="./images/5026416_s.jpg"
    property="og:image"
  />,
  <meta
    content="#00a300"
    name="msapplication-TileColor"
  />,
  <meta
    content="/favicon/browserconfig.xml"
    name="msapplication-config"
  />,
  <meta
    content="#fff"
    name="theme-color"
  />,
  <link
    href="/favicon/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
  <link
    href="/favicon/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="/favicon/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />,
  <link
    href="/favicon/favicon.ico"
    rel="icon"
  />,
  <link
    href="/favicon/site.webmanifest"
    rel="manifest"
  />,
  <link
    color="#5bbad5"
    href="/favicon/safari-pinned-tab.svg"
    rel="mask-icon"
  />,
  <link
    href="/feed.xml"
    rel="alternate"
    type="application/rss+xml"
  />,
  <link
    href="https://food-blog-chi.vercel.app/articles/p97vmuno3jdn"
    rel="canonical"
  />,
]
`;
